packages/flutter_tools/lib/src/asset.dart

```diff
@@ -77,14 +77,28 @@ enum AssetKind {
   model,
 }
 
-abstract class AssetBundle {
-  Map<String, DevFSContent> get entries;
+/// Contains all information about an asset needed by tool the to prepare and
+/// copy an asset file to the build output.
+@immutable
+final class AssetBundleEntry {
+  const AssetBundleEntry(this.content, {
+    this.kind = AssetKind.regular,
```

Comment:
`AssetKind.regular` was used as a default in a few different places, so I decided to make it a default here.

Reply:
can we at least add a dartdoc to `AssetKind.regular`--I don't personally know what a "regular" asset is.

Reply:
Actually, the more I think about this, the more I worry we introduce bugs where we accidentally let this fall back `AssetKind.regular`. Can you convince me otherwise?

Reply:
The problem is that a "regular" asset is simply what it _isn't_—a font, shader, or model. I can document that, but I don't know if negative definitions are particularly helpful.

Reply:
> Actually, the more I think about this, the more I worry we introduce bugs where we accidentally let this fall back `AssetKind.regular`. Can you convince me otherwise?

No, to be honest. Default parameters always risk this as a tradeoff for less noise throughout the rest of the code. I'm happy to make it required. I originally opted not to for the initial PR review since it would have made the diff more intimidating.

Reply:
> The problem is that a "regular" asset is simply what it isn't—a font, shader, or model. I can document that, but I don't know if negative definitions are particularly helpful.

Interesting, so it basically means "just copy, don't do any pre-processing"?

Reply:
also, what's a model?

Reply:
> Interesting, so it basically means "just copy, don't do any pre-processing"?

Yeah, my hope is that the docstring for `AssetBundleEntry` alludes-to/implies this. 

As an example, you can see this stuff in [devfs.dart](https://github.com/flutter/flutter/blob/c1107a9956be4cc3114a293f451bc32c48edc0cb/packages/flutter_tools/lib/src/devfs.dart#L638).

> also, what's a model?

A file containing data for a 3D model, I believe.